katello/katello-change-hostname

```diff
@@ -0,0 +1,225 @@
+installation_media_paths = `hammer -u #{options[:username]} -p #{options[:password]} medium list | grep "#{old_hostname}" | awk '{ print $5 }'`.split("\n")
+installation_media_ids.zip(installation_media_paths).each do |id, path|
+  `hammer -u #{options[:username]} -p #{options[:password]} medium update --id #{id} --path #{path.sub! old_hostname, new_hostname}`   
+end
+
+if is_rhel_version?("6")
+  STDOUT.puts "updating hostname in /etc/sysconfig/network"
+  STDOUT.puts("sed -i -e 's/#{old_hostname}/#{new_hostname}/g' /etc/sysconfig/network")
+  `sed -i -e 's/#{old_hostname}/#{new_hostname}/g' /etc/sysconfig/network`
+  STDOUT.puts "setting hostname"
+  `hostname #{new_hostname}`
+elsif is_rhel_version?("7")
+  STDOUT.puts "updating hostname in /etc/hostname"
+  `sed -i -e 's/#{old_hostname}/#{new_hostname}/g' /etc/hostname`
+  STDOUT.puts "setting hostname"
+  `hostnamectl set-hostname #{new_hostname}`
+else
+  STDOUT.puts "couldn't find RHEL version: #{RHEL_VERSION} - exiting, no changes have been made to your system"
+  exit
+end
+
+STDOUT.puts "checking if hostname was changed"
+if get_hostname != new_hostname
+  STDOUT.puts "the new hostname was not changed successfully, exiting script"
+  exit
+end
+
+STDOUT.puts "stopping services"
+`katello-service stop`
+
+STDOUT.puts "deleting old certs"
+`
+rm -rf #{katello_answers["certs"]["pki_dir"]}{,.bak}
+rm -rf /etc/pki/katello-certs-tools{,.bak}
+rm -rf /etc/candlepin/certs/amqp{,.bak}
+rm -rf #{katello_answers["foreman_proxy"]["ssl_ca"]}
+rm -rf #{katello_answers["foreman_proxy"]["ssl_cert"]}
+rm -rf #{katello_answers["foreman_proxy"]["ssl_key"]}
+rm -rf #{katello_answers["foreman_proxy"]["foreman_ssl_ca"]}
+rm -rf #{katello_answers["foreman_proxy"]["foreman_ssl_cert"]}
+rm -rf #{katello_answers["foreman_proxy"]["foreman_ssl_key"]}
+rm -rf /etc/foreman/old-certs
+rm -rf #{katello_answers["foreman"]["client_ssl_ca"]}
+rm -rf #{katello_answers["foreman"]["client_ssl_cert"]}
+rm -rf #{katello_answers["foreman"]["client_ssl_key"]}
+rm -rf #{katello_answers["foreman_proxy"]["ssldir"]}
+rm -rf /etc/pki/katello/nssdb
+rm -f /etc/tomcat/keystore
+rm -f /etc/pki/katello/keystore
+cp -r #{ssl_build_dir} #{ssl_backup_dir}
+rm -rf #{ssl_build_dir}
+`
+STDOUT.puts "backed up #{ssl_build_dir} to #{ssl_backup_dir}"
+STDOUT.puts "updating hostname in /etc/hosts"
+`sed -i -e 's/#{old_hostname}/#{new_hostname}/g' /etc/hosts`
+
+STDOUT.puts "updating hostname in foreman installer scenarios"
+`sed -i -e 's/#{old_hostname}/#{new_hostname}/g' /etc/foreman-installer/scenarios.d/*.yaml`
+
+STDOUT.puts "removing last_scenario.yml file"
+`rm -rf /etc/foreman-installer/scenarios.d/last_scenario.yaml`
+
+STDOUT.puts "re-running the installer"
+
+installer = "#{options[:program]}-installer --scenario #{options[:scenario]} -v --certs-regenerate-ca=true --certs-regenerate=true \
```

Comment:
If my setup has custom certs will this replace them? If so can we do one of the follow:
- Notify the user to re install custom certs with `foreman-installer <blah>` when we're done (maybe check and notify them before we do anything as well)
- Allow user to pass new custom certs in
- Someother better solution you can think of ;)